pylint_odoo/misc.py

```diff
@@ -112,14 +113,37 @@ def get_manifest_file(self, node_file):
 
     def set_ext_files(self):
         """Create `self.ext_files` dictionary with {extension_file: [files]}
+            and exclude files using --ignore and --ignore-patterns parameters
         """
         self.ext_files = {}
+        base_path = os.path.dirname(os.path.realpath(__file__))
+        ignores = [os.path.realpath(os.path.join(base_path, ignore))
+                   for ignore in self.linter.option_value('ignore')]
+        ignores_patterns = [glob.glob(os.path.join(base_path, ignore.pattern))
+                            for ignore
+                            in self.linter.option_value('ignore-patterns')]
+        for ignore in ignores_patterns:
+            for path in ignore:
+                ignores.append(path)
         for root, _, filenames in os.walk(self.module_path, followlinks=True):
-            for filename in filenames:
-                fext = os.path.splitext(filename)[1].lower()
-                fname_rel = os.path.relpath(
-                    os.path.join(root, filename), self.module_path)
-                self.ext_files.setdefault(fext, []).append(fname_rel)
+            find = True
+            for ignore in ignores:
+                if ignore in root:
+                    find = False
+                    continue
+            if find:
```

Comment:
The cyclomatic complexity of this method is too high.
Could you refactoring it to reduce it?